optuna/multi_objective/samplers/_motpe.py

```diff
@@ -0,0 +1,649 @@
+        consider_magic_clip:
+            Enable a heuristic to limit the smallest variances of Gaussians used in
+            the Parzen estimator.
+        consider_endpoints:
+            Take endpoints of domains into account when calculating variances of Gaussians
+            in Parzen estimator. See the original paper for details on the heuristics
+            to calculate the variances.
+        n_startup_trials:
+            The random sampling is used instead of the MOTPE algorithm until the given number
+            of trials finish in the same study. 11 * number of variables - 1 is recommended in the
+            original paper.
+        n_ehvi_candidates:
+            Number of candidate samples used to calculate the expected hypervolume improvement.
+        gamma:
+            A function that takes the number of finished trials and returns the number of trials to
+            form a density function for samples with low grains. See the original paper for more
+            details.
+        weights:
+            A function that takes the number of finished trials and returns a weight for them. As
+            default, weights are automatically calculated by the MOTPE's default strategy.
+        seed:
+            Seed for random number generator.
+
+    The original paper utilizes Latin hypercube sampling (LHS) for initialization. The following
+    code is an example of initializing MOTPE using LHS. Here
+    `pyDOE2 <https://pypi.org/project/pyDOE2/>`_ is used for LHS.
+
+    Example:
+
+        .. testcode::
+
+                import numpy as np
+                import pyDOE2
+
+                seed = 128
+                num_variables = 9
+                n_startup_trials = 11 * num_variables - 1
+
+                def objective(trial):
+                    x = []
+                    for i in range(1, num_variables + 1):
+                        x.append(trial.suggest_uniform(f"x{i}", 0.0, 2.0 * i))
+                    return x
+
+                sampler = optuna.multi_objective.samplers.MOTPEMultiObjectiveSampler(
+                    n_startup_trials=n_startup_trials,
+                    n_ehvi_candidates=24,
+                    seed=seed
+                )
+                # NOTE: Sampled vectors are normalized. Denormalization is needed when a search
+                # space is not normalized.
+                initial_samples = pyDOE2.lhs(
+                    num_variables, samples=n_startup_trials, criterion='maximin',
+                    random_state=np.random.RandomState(seed)
+                )
+                for s in initial_samples:
+                    study.enqueue_trial({f'x{i}': v for i, v in enumerate(s, start=1)})
+
+                study.optimize(objective, n_trials=250)
+    """
+
+    def __init__(
+        self,
+        consider_prior: bool = True,
+        prior_weight: float = 1.0,
+        consider_magic_clip: bool = True,
+        consider_endpoints: bool = True,
```

Comment:
The original MOTPE's Parzen estimator implementation follows the original TPE's Parzen estimator, i.e., consider_endpoints = True.
Therefore, currently, this option is enabled as default.
However, if you want to ensure consistency with `TPESampler`, I will change the default of this option to False.
The decision can be made by the Optuna team.

Reply:
Thank you for your explanation. That makes sense.
IMO, we can use different values since the two algorithms are not the same.

@HideakiImamura Do you have any comments?

Reply:
I'm fine with the current implementation. @y0z Thank you for your detailed explanation.

Reply:
Let me resolve this conversation since the discussion seems to be settled.